### PR TITLE
Add key generation

### PR DESCRIPTION
**Pull request by kcking**
move KeyType to crate::key since it is supported on iOS

### CODE REVIEW

**Comment by kornelski** [comment]
Looks good!

File: security-framework/src/key.rs [hunk 1]
```diff
@@ -28,7 +94,58 @@ impl_TCFType!(SecKey, SecKeyRef, SecKeyGetTypeID);
 unsafe impl Sync for SecKey {}
 unsafe impl Send for SecKey {}
 
+/// Which Keychain to store the generated key in. On iOS this struct is empty as
+/// there is only one keychain.
+pub struct StoreInKeychain {
+    #[cfg(target_os="macos")]
```
Comment (kornelski) [inline security-framework/src/key.rs:100]: maybe `generate` could rely on taking `None` instead of empty struct?

**Comment by kcking** [comment]
Thanks for taking a look! I'm still iterating on the implementation a bit as I use it in a project. Hoping to support secure enclave keys as well :)

**Comment by kcking** [comment]
I am breaking this long chain of commits down into multiple PRs, the first of which is https://github.com/kornelski/rust-security-framework/pull/164

**Pull request closed without merging**